### PR TITLE
Grid collection

### LINKED ISSUE #217: ENH: block collection subclass to directly load a list of grids?
In running the spherical renderer on a real dataset, I had to do some massaging to get a decent grid decomposition. That would have been a fair bit easier if I could instead have just supplied a list of grid objects (or maybe a list of covering or arbitrary grid objects?). I think there might be some value in adding a method for loading non-overlapping level 0 grids without the block traversal kicking in? This would come with the danger that a user supply grids that are too large to load... but that was happening to me with the default behavior when I tried to load a dataset that has a single large grid. While it'd be worth improving how yt decomposes a single large level 0 grid, I think yt_idv would also benefit from having a way to load an arbitrary list of grid objects.

### PR DESCRIPTION
**Pull request by chrishavlin**
This PR:

* refactors the block collection to create an abstract base class for collections 
* adds a GridCollection object for directly rendering a list of covering or arbitrary grid objects

Close #217 

To do:

- [ ] add some figures to this PR
- [ ] update docs
- [ ] get the block outlines working for GridCollection

### CODE REVIEW

**Comment by chrishavlin** [comment]
the cartesian covering grid test image is empty

**Comment by chrishavlin** [comment]
grid outlines working -- also removed the block outlines menu item for this rendering object since they are not applicable. 

snap_0000

**Comment by chrishavlin** [comment]
oh, hit undraft too soon maybe. forgot i need to double check the plain covering_grid behavior

File: yt_idv/scene_data/block_collection.py [hunk 1]
```diff
@@ -351,3 +380,46 @@ def _block_collection_outlines(
...
+    def _get_volume_data(self, block):
+        return block.grid[self._field]
+
+    def _get_ds(self):
+        return self.data_source[0].ds
+
+    def viewpoint_iter(self, camera):
+        # neglecting camera
+        for vbo_i in range(len(self.blocks)):
+            yield (vbo_i, self.texture_objects[vbo_i], self.bitmap_objects[vbo_i])
```
Comment (chrishavlin) [inline yt_idv/scene_data/block_collection.py:421]: So this i the one bit I couldn't figure out easily: this will just return all the grid objects without any along-ray sorting... which is fine for max intensity (and projective?) projections. but will result in incorrect results for a full transfer function.

**Comment by chrishavlin** [comment]
well, very confused about why the cartesian covering grid test case saves an empty image... running the same code locally with osmesa returns the expected figure 

grid_coll_offscreen_cartesian_covering

it's only when running it in pytest that it is empty.
